### PR TITLE
RDS, S3 integration for KF-v1.4

### PR DESCRIPTION
**Pull request by surajkota**
### Description of your changes:

All AWS related changes are confined to `distributions/aws` directory with this PR

- RDS and S3 integration for pipelines: 
  - ported `aws` directory from v1.3-branch under `apps/pipeline/upstream/env` to `distributions/aws/apps/pipeline`
  - updated the base resources location in `distributions/aws/apps/pipeline/kustomization.yaml`
- RDS integration for Katib: 
   - ported `katib-external-db-with-kubeflow` directory from v1.3-branch directory under `apps/katib/upstream/installs` to `distributions/aws/apps`
   - updated base resources location in `distributions/aws/apps/katib-external-db-with-kubeflow/kustomization.yaml` and created a patches directory for db-manager patch. `db-manager.yaml` patch is a copy from `katib-external-db` because patches need to be under kustomization directory
- DLC based jupyter notebooks
  - created `jupyter-web-app` directory under `distributions/aws/apps` and created an overlay with spawner_ui_config with AWS DLC based images for tensorflow and pytorch. spawner_ui_config is a copy of the file `apps/jupyter/jupyter-web-app/upstream/base/configs/spawner_ui_config.yaml` except for notebook server images updates
- Changes from #20 to v1.4-branch. No changes here
 
### Testing
Manual, WIP
Deleted generic components installed in an existing 1.4 deployment and installed aws modified components
- [x] Pipelines: Ran sample data passing pipeline, verified databases and run detail in RDS and artifacts pushed in S3
- [x] Notebook: created notebook server with tf-cpu image and ran some commands
- [x] Katib: obseravtion_logs table was created by db-manager so assuming the connection is good. test run will be addressed as part of #30 

#27

### CODE REVIEW

File: distributions/aws/apps/jupyter-web-app/configs/spawner_ui_config.yaml [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+        # Note that this is a templated value. Special values:
+        # {notebook-name}: Replaced with the name of the Notebook. The frontend
+        #                  will replace this value as the user types the name
+        value: 'workspace-{notebook-name}'
+      size:
+        # The Size of the Workspace Volume (in Gi)
+        value: '5Gi'
+      mountPath:
+        # The Path that the Workspace Volume will be mounted
+        value: /home/jovyan
```
Comment (goswamig) [inline distributions/aws/apps/jupyter-web-app/configs/spawner_ui_config.yaml:96]: Whats this?

**Review by goswamig — COMMENTED** [review]
Can you add the details on testing done on this? 
Can you point out any differences w.r.t. 1.3?

File: distributions/aws/apps/jupyter-web-app/configs/spawner_ui_config.yaml (hunk 1 above)
Comment (surajkota) [reply distributions/aws/apps/jupyter-web-app/configs/spawner_ui_config.yaml:96]: path where PV created for the notebook server is mounted to the server pod.

the only change in this file is the notebook-servers ecr images. Rest is same as [upstream](https://github.com/awslabs/kubeflow-manifests/blob/v1.4-branch/apps/jupyter/jupyter-web-app/upstream/base/configs/spawner_ui_config.yaml)

**Review by goswamig — APPROVED** [review]

**Comment by goswamig** [comment]
Lets schedule a bug bash to run through it.

**Review by rrrkharse — APPROVED** [review]

**Review by mbaijal — APPROVED** [review]
Reviewed the notebook section, looks fine assuming you have already tested to check you see the right default options and are able to spin up a notebook server. 

I will create a separate PR on both branches to fix the known issue we have.

**Pull request merged**